openlibrary/plugins/upstream/models.py

```diff
@@ -15,21 +15,25 @@
 from infogami.utils import stats
 
 from openlibrary.core import models, ia
-from openlibrary.core.models import Image
+from openlibrary.core.models import Thing, Image
 from openlibrary.core import lending
 
 from openlibrary.plugins.upstream.utils import MultiDict, parse_toc, get_edition_config
 from openlibrary.plugins.upstream import account
 from openlibrary.plugins.upstream import borrow
-from openlibrary.plugins.worksearch.code import works_by_author
+from openlibrary.plugins.worksearch.code import SearchResponse, works_by_author
 from openlibrary.plugins.worksearch.search import get_solr
 
 from openlibrary.utils import dateutil
 from openlibrary.utils.isbn import isbn_10_to_isbn_13, isbn_13_to_isbn_10
 from openlibrary.utils.lccn import normalize_lccn
+from infogami.infobase.client import Nothing
+from openlibrary.plugins.upstream.models import Edition
+from typing import Any, Dict, List, Optional, Union
+from web.utils import Storage
 
 
-def follow_redirect(doc):
+def follow_redirect(doc: Union[Thing, Author]) -> "Author":
```

Comment:
???
```suggestion
def follow_redirect(doc: Union[Thing, models.Author]) -> "Author":
```
or remove the type hint if it is more work that it is worth.

Reply:
I have been running into the circular import thing fairly frequently, but so far with a combination of:
```
if TYPE_CHECKING:
    <import the upstream stuff>
```
and then putting quotes around the things to make them strings, such as `"Author"` has above, seems to be working.

Reply:
I think that importing `models.Author` and `models.Edition` and defining local classes `Author` and `Edition` is better.